### PR TITLE
Wysiwyg Menu Improvements

### PR DESCRIPTION
**Pull request by spbyrne**
* Uses flexbox to wrap menu items when needed
* Adds highlighting for current heading level in DDL
* Strips heading styles so page styles are used for heading DDL
* Reduces CSS repetition 
* Adds menu portal & uses it for heading DDL

### CODE REVIEW

File: packages/tinacms/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,14 @@
...
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+# [0.14.0](https://github.com/tinacms/tinacms/compare/tinacms@0.14.0-alpha.1...tinacms@0.14.0) (2020-02-24)
+
+### Bug Fixes
+
+- **sidebar:** fix toggling hidden state ([254326c](https://github.com/tinacms/tinacms/commit/254326c))
+
+### Features
+
```
Comment (ncphillips) [inline packages/tinacms/CHANGELOG.md]: Accidental duplication

```suggestion
```

File: packages/@tinacms/fields/src/Wysiwyg/state/plugins/Menu/Menu.tsx [hunk 2]
```diff
@@ -112,13 +113,13 @@ export const Menu = (props: Props) => {
...
+      const menuWrapperDiv = menuRef.current.parentElement
+      const wysiwygEditorDiv = menuWrapperDiv.nextSibling
+      const startPosition = menuRef.current ? menuWrapperDiv.offsetTop : 0
       const endPosition = menuRef.current
-        ? menuRef.current.parentElement.nextSibling.offsetHeight +
-          menuRef.current.parentElement.offsetHeight +
-          menuRef.current.parentElement.offsetTop
+        ? wysiwygEditorDiv.offsetHeight +
+          menuWrapperDiv.offsetHeight +
+          menuWrapperDiv.offsetTop
```
Comment (ncphillips) [inline packages/@tinacms/fields/src/Wysiwyg/state/plugins/Menu/Menu.tsx]: Nice names 👍

File: packages/@tinacms/fields/src/Wysiwyg/state/plugins/Menu/FormattingDropdown.tsx [hunk 3]
```diff
@@ -101,41 +106,30 @@ function makeToggleHeader(level: number) {
...
-const HeadingSix = styled.div`
-  font-size: 16px;
-  white-space: nowrap;
-  line-height: 1;
-  display: block;
+const HeadingFive = styled.h5`
+  ${BaseHeading}
+`
+const HeadingSix = styled.h6`
+  ${BaseHeading}
```
Comment (jpuri) [inline packages/@tinacms/fields/src/Wysiwyg/state/plugins/Menu/FormattingDropdown.tsx]: 👍

File: packages/@tinacms/react-core/CHANGELOG.md [hunk 4]
```diff
@@ -7,165 +7,85 @@ See [Conventional Commits](https://conventionalcommits.org) for commit guideline
...
 
-
 ### Features
 
-* **useGlobalForm:** add hook for registering global forms ([d450cae](https://github.com/tinacms/tinacms/commit/d450cae))
-
-
-
-
+- **useGlobalForm:** add hook for registering global forms ([d450cae](https://github.com/tinacms/tinacms/commit/d450cae))
```
Comment (jpuri) [inline packages/@tinacms/react-core/CHANGELOG.md]: Not sure if this file should be there ?

**Comment by jpuri** [comment]
This file looks like not needed in the PR: packages/react-tinacms-inline/package-lock.json

**Comment by jpuri** [comment]
New flexible menu works like charm 🎉

File: packages/@tinacms/react-core/CHANGELOG.md (hunk 4 above)
Comment (spbyrne) [reply packages/@tinacms/react-core/CHANGELOG.md]: This came from an odd merge but I've had a couple sets of eyes on it and we think everything looks okay.

**Pull request merged**